aten/src/ATen/DLConvertor.cpp

```diff
@@ -101,8 +101,9 @@ static DLDevice getDLDevice(const Tensor& tensor, const int64_t& device_id) {
       ctx.device_type = DLDeviceType::kDLROCM;
       break;
     case DeviceType::XPU:
-      ctx = at::detail::getXPUHooks().getDLPackDeviceFromATenDevice(
-          ctx, tensor.device(), tensor.data_ptr());
+      ctx.device_type = DLDeviceType::kDLOneAPI;
+      ctx.device_id =
+          at::detail::getXPUHooks().getGlobalIdxFromDevice(tensor.device());
```

Comment:
`kDLOneAPI` is a device type from DLPack, XPU uses it here. To avoid leaking DLPack implementation detail to every project that includes `ATen/Context.h`, we decouple `DLDevice_` from `XPUHooks` and only use the index of `DLDevice_` as our input.